sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/BlobProperties.java

```diff
@@ -191,6 +194,116 @@ public BlobProperties(final OffsetDateTime creationTime, final OffsetDateTime la
         this.committedBlockCount = committedBlockCount;
         this.versionId = versionId;
         this.isCurrentVersion = isCurrentVersion;
+
+        this.objectReplicationSourcePolicies = new HashMap<>();
+        this.objectReplicationDestinationPolicyId = objectReplicationStatus.getOrDefault("policy-id", null);
```

Comment:
Is there a clean way we could unduplicate this code? I'm not sure where it could live though.

Reply:
I don't think so. One lives in the readme and the other lives in BlobProperties. And they aren't totally identical because the one here saves this as the actual property rather than constructing it in the getter, and that would be way more complicated to do in swagger.

Reply:
Even with the update of moving this out of the readme, I don't know that there's a great place for us to put this code to get rid of the duplication. There's not really a common code path between them, and we're trying to move away from using utility methods.